sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -605,13 +610,21 @@ def values(self) -> ValuesView[Union[str, Mapping[str, Any]]]:
         resolved.
 
         :return: A list of values loaded from Azure App Configuration. The values are either Strings or JSON objects,
-        based on there content type.
+         based on there content type.
         :rtype: ValuesView[Union[str, Mapping[str, Any]]]
         """
         with self._update_lock:
             return copy.deepcopy((self._dict)).values()
 
-    def get(self, key: str, default: Optional[str] = None) -> Union[str, Mapping[str, Any], None]:
+    @overload
+    def get(self, key: str, default: None = None) -> Union[str, JSON, None]:
+        ...
+
+    @overload
+    def get(self, key: str, default: Union[str, JSON, _T]) -> Union[str, JSON, _T]:  # pylint: disable=signature-differs
+        ...
+
+    def get(self, key: str, default: Optional[Union[str, JSON, _T]] = None) -> Union[str, JSON, _T, None]:
```

Comment:
Since we inherit from a Mapping, mypy insists that are overloads match the base impl: https://github.com/python/typeshed/blob/main/stdlib/typing.pyi#L628